### PR TITLE
Resolve permission issue when updating submodules

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Switches to Commander token to resolve the push access when updating submodules
Starts to update only the repo/submodule that triggered the pipeline 

## Why?

Submodules update pipeline fails with permission issue

## Checklist

1. Closes 

2. How was this tested:

not tested

3. Any docs updates needed?

### CODE REVIEW

**Review by dnr — COMMENTED** [review]
Can you remove the TOKEN from the "Commit update" step, then? If wherever this persistent token thing stores the token is enough to mess up the commit step with an explicit token, then that one should be able to just use this, right?

**Comment by feedmeapples** [comment]
@dnr it seems necessary to be set explicitly when making a push into a protected branch

**Comment by feedmeapples** [comment]
> I'm confused.. can you remove the other usage of the token or not?

it seems necessary to be set explicitly when making a push into a protected branch

File: .github/workflows/update-submodules.yml [hunk 1]
```diff
@@ -49,8 +49,13 @@ jobs:
           SUBMODULE: ${{ github.event.inputs.repo }}
           BRANCH: ${{ steps.target_branches.outputs.submodule_branch }}
         run: |
+          if [[ "$SUBMODULE" != "temporal" && "$SUBMODULE" != "tctl" ]]
```
Comment (dnr) [inline .github/workflows/update-submodules.yml:52]: Maybe do:
```
if ! git config --file=.gitmodules --get submodule."$SUBMODULE".url >/dev/null; then
  echo "unknown submodule $SUBMODULE"
  exit 1
fi
```
so it's using the same source of truth about which submodules exist?

**Review by dnr — APPROVED** [review]
Okay. One more suggestion:

**Comment by feedmeapples** [comment]
> I'm confused.. can you remove the other usage of the token or not?

@dnr is was required yes. I've went through the docs again to see exactly why this works this way. Also went checking if "Allow specified actors to bypass pull request requirements" would affect the need to set the token to a remote. So far seems it's just required at least with https and not ssh

**Pull request merged**